src/Models/AzureFunctions.ts

```diff
@@ -185,6 +208,11 @@ export class AzureFunctions implements Component, Provisionable, Deployable {
   }
 
   async provision(): Promise<boolean> {
+    const isFunctionsExtensionAvailable = await AzureFunctions.isAvailable();

```

Comment:
Is it possible to check whether the project contains azure function and then ask user to install it?

Reply:
Yeah. `AzureFunctions.isAvailable` is a static method, so we can call it anywhere without context.

I thinks this is more like a user experience topic. Should we show the notification as soon as the user opens a project contains Functions component or only when user executes provision command?